### PR TITLE
Add OpenShift CI for rhmdnd/compserv

### PR DESCRIPTION
**Pull request by rhmdnd**

### CODE REVIEW

**Comment by rhmdnd** [comment]
Just FYI - the owners check failed since the invitations for the CI bots are still pending.

**Comment by rhmdnd** [comment]
CI failed because the compserv repository doesn't have an OWNERS file. Fixed in https://github.com/rhmdnd/compserv/pull/42

**Comment by jhrozek** [comment]
So I don't know if/how much things had changed recently, but openshift used to mandate vendoring for all repos under the openshift org. Reading the error messages, it might be enough to explictily add `-mod=readonly`? But either way, please run `go mod tidy`, I noticed that there's requirements in `go.mod` that are not needed anymore.

File: ci-operator/config/rhmdnd/compserv/rhmdnd-compserv-main.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+binary_build_commands: go build -o builds/ cmd/compserv-server.go
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.16
```
Comment (jhrozek) [inline ci-operator/config/rhmdnd/compserv/rhmdnd-compserv-main.yaml:6]: I think it should be fine to start with something more modern.

**Comment by rhmdnd** [comment]
Proposed an update of the dependencies and rerunning the tests to get a fresh run.

https://github.com/rhmdnd/compserv/pull/44

File: ci-operator/config/rhmdnd/compserv/rhmdnd-compserv-main.yaml (hunk 1 above)
Comment (rhmdnd) [reply ci-operator/config/rhmdnd/compserv/rhmdnd-compserv-main.yaml]: Addressed this in compserv https://github.com/rhmdnd/compserv/pull/47

**Review by Vincent056 — APPROVED** [review]
/lgtm

**Pull request merged**